demos/common/azure_ca_recovery/azure_ca_recovery_parse.c

```diff
@@ -0,0 +1,205 @@
+    {
+        RETURN_IF_JSON_TOKEN_NOT_TYPE( ref_json_reader, AZ_JSON_TOKEN_PROPERTY_NAME );
+
+        if( az_json_token_is_text_equal(
+                &ref_json_reader->token,
+                AZ_SPAN_FROM_STR( AZ_IOT_CA_RECOVERY_SIGNATURE_NAME ) ) )
+        {
+            _az_RETURN_IF_FAILED( az_json_reader_next_token( ref_json_reader ) );
+            RETURN_IF_JSON_TOKEN_NOT_TYPE( ref_json_reader, AZ_JSON_TOKEN_STRING );
+
+            if( ref_json_reader->token.kind != AZ_JSON_TOKEN_NULL )
+            {
+                recovery_payload->payload_signature = ref_json_reader->token.slice;
+            }
+        }
+        else if( az_json_token_is_text_equal(
+                     &ref_json_reader->token,
+                     AZ_SPAN_FROM_STR( AZ_IOT_CA_RECOVERY_CERT_TRUST_BUNDLE_NAME ) ) )
+        {
+            _az_RETURN_IF_FAILED( az_json_reader_next_token( ref_json_reader ) );
+            RETURN_IF_JSON_TOKEN_NOT_TYPE( ref_json_reader, AZ_JSON_TOKEN_STRING );
+
+            if( ref_json_reader->token.kind != AZ_JSON_TOKEN_NULL )
+            {
+                recovery_payload->trust_bundle_json_object_text = ref_json_reader->token.slice;
+            }
+        }
+
+        _az_RETURN_IF_FAILED( az_json_reader_next_token( ref_json_reader ) );
+    }
+
+    return AZ_OK;
+}
+
+/* END EMBEDDED SDK CODE SECTION */
+
+static void prvCastCoreToMiddleware( AzureIoTCARecovery_RecoveryPayload * pxRecoveryPayload,
+                                     azure_iot_ca_recovery_payload * recovery_payload )
+{
+    pxRecoveryPayload->pucPayloadSignature = az_span_ptr( recovery_payload->payload_signature );
+    pxRecoveryPayload->ulPayloadSignatureLength = az_span_size( recovery_payload->payload_signature );
+    pxRecoveryPayload->pucTrustBundleJSONObjectText = az_span_ptr( recovery_payload->trust_bundle_json_object_text );
+    pxRecoveryPayload->ulTrustBundleJSONObjectTextLength = az_span_size( recovery_payload->trust_bundle_json_object_text );
+
+    pxRecoveryPayload->xTrustBundle.pucCertificates = az_span_ptr( recovery_payload->trust_bundle.certificates );
+    pxRecoveryPayload->xTrustBundle.ulCertificatesLength = az_span_size( recovery_payload->trust_bundle.certificates );
+    pxRecoveryPayload->xTrustBundle.ulVersion = recovery_payload->trust_bundle.version;
+    pxRecoveryPayload->xTrustBundle.ullExpiryTimeSecs = recovery_payload->trust_bundle.expiry_time;
+}
+
+AzureIoTResult_t AzureIoTCARecovery_ParseRecoveryPayload( AzureIoTJSONReader_t * pxReader,
+                                                          AzureIoTCARecovery_RecoveryPayload * pxRecoveryPayload )
+{
+    az_result xCoreResult;
+    azure_iot_ca_recovery_payload recovery_payload;
+    az_json_reader jr;
+
+    xCoreResult = az_iot_ca_recovery_parse_recovery_payload( &pxReader->_internal.xCoreReader, &recovery_payload );
+
+    if( az_result_failed( xCoreResult ) )
+    {
+        /* TODO: Uncomment once this is a real API (is a private func) */
```

Comment:
unfortunately not. since these don't live in the SDK, we don't have access to the `private` header. We could maybe hack our way by hardcoding the include, but that's not really clean :/